AppVeyor: Run only SqlServer FunctionalTests

smitpatel opened the pull request:
This PR enables only SqlServer Functional Tests on AppVeyor. It takes about ~23 minutes for 1 run to finish.
Since we already had made groups of testsin #7842 this PR just run the group which travis cannot run.

Also had to change to SqlServer 2016 because for some reason SqlServer 2014 is not starting up in AppVeyor.

smitpatel commented:
Facing timeout errors. putting in WIP.

Eilon commented on appveyor.yml:
```diff
@@ -1,16 +1,13 @@
-only_commits:
-  # Start a build if commit message contains 'appveyor'
-  message: /appveyor/
```
Yay!

divega commented on appveyor.yml:
```diff
@@ -1,16 +1,13 @@
 # See http://www.appveyor.com/docs/services-databases
 services:
-  - mssql2014
+  - mssql2016
```
Does anyone know what exact version is this? Is it regular SQL Server or LocalDB, is it RTM or RC1? Curious because that can affect whether [instant database file initialization](https://msdn.microsoft.com/en-us/library/ms175935.aspx) can and will be enabled.

smitpatel replied:
`SQL Server 2016 Express`

smitpatel commented:
This is ready for review.
Changes introduced
- Run AppVeyor on all PRs but only run SqlServer FunctionTests.
- I have preserved grouping Nate did. Though our basic requirement is SqlServer & non-SqlServer tests only.
- This does not run benchmark tests. It involves importing adventureworks in AppVeyor before calling build. I believe if all other tests are passing then benchmark would not fail.

I monitored different issues happening when AppVeyor was failing. The biggest issue was error with sql2014 service. For some reasons (associated with AppVeyor), it fails to start instance of sql server 2014. That explains the consistent failures we were having for past few months. I changed it to use Sql Server 2016 now which is working fine.

There were also timeout issues happening. I also noticed those happening on efci when I removed the environment variable for No parallel tests. Ideally the assembly condition should be sufficient enough but it doesn't seem to be the case. (I am still monitoring EFCI to check it).
Changes in #7877 fixes the timeout issues. This has been consistently passing in last 3 runs. It takes about 20 mins to finish so we don't hog up queues either.

smitpatel commented:
We do have `sqlcmd` available on AppVeyor. I am checking about SQLCLient ODBC drivers installation,

ajcvickers commented:
So, with this change, the plan is to have AppVeyor run the SQL Server tests, and have Travis for everything else?

smitpatel commented:
@ajcvickers - Yes. For PRs on query this can be very useful because sqlserver functional tests are more likely to fail.

bricelam commented on build/repo.targets:
```diff
@@ -0,0 +1,40 @@
+      <ProjectsToTest Include="@(UnitTests)" Condition="'$(TestGroup)' == 'Unit'" />
+      <ProjectsToTest Include="@(FunctionalTests)" Condition="'$(TestGroup)' == 'Functional'" />
+      <ProjectsToTest Include="@(BenchmarkTests)" Condition="'$(TestGroup)' == 'Benchmarks'" />
+      <ProjectsToTest Include="@(SqlServerTests)" Condition="'$(TestGroup)' == 'SqlServer'" />
```
Only `SqlServer` is being used. Remove `All`, `Unit`, `Functional` & `Benchmark` groups until we need them.

bricelam replied:
(Their items too)

smitpatel replied:
Updated.
